unic/char/property/src/domain.rs

```diff
@@ -16,6 +16,19 @@ use std::fmt::{Debug, Display};
 use std::hash::Hash;
 
 
+/// A Character Property, defined for some or all Unicode characters.
+pub trait CharProperty: PartialCharProperty {
```

Comment:
We could even keep the `impl PartialCharProperty for CompleteCharProperty` if we wanted, but the advantage of having them be disjoint is that importing both into the namespace will still allow unambiguous `::of()` calls.